### PR TITLE
homework 4 all tasks

### PR DESCRIPTION
**Pull request by AlexandrBelan**
homework 4 all tasks comleted

### CODE REVIEW

File: JS_lesson4/Task2.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// 2. Преобразовать написанный код в task 1 так, чтобы сначала проверялся тип данных. 
+//   И если он не number - кидалась ошибка в консоль.
+//   Проверить работу кода на следующих данных 17, 18, 61, "2", "aaa"
+// 3. Преобразовать Task 2 - 1 таким образом, чтобы значение НАПРИМЕР '2' (т.е. ЛЮБАЯ строка в которой лежат ТОЛЬКО ЦИФРЫ) пропускалось, 
+//   преобразовываясь в number
+const age = "54";
+const age_2 = 18;
+const age_3 = 60;
+    switch (true) {
+        case Number.isNaN(Number(age)): {
```
Comment (anatoly-karpovich) [inline JS_lesson4/Task2.js]: I belive this case should be under IF statement BEFORE switch to cut invalid data. And then to work in switch with only valid numbers

File: JS_lesson4/Task2.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            throw new Error('Parameter is not a number!');
+            break;
+        } 
+        case Number(age) < age_2: {
+            console.log(`You dont have access cause your age is ${age} Its less then`);
+            break;
+        }
+        case Number(age) >= age_2 && Number(age) < age_3: {
```
Comment (anatoly-karpovich) [inline JS_lesson4/Task2.js]: Number(age) is duplicated in several rows. It is better to place this convertation BEFORE switch block

File: JS_lesson4/Task3.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// Преобразовать Task 2* таким образом, чтобы age принимался через prompt в привязанной вёрстке, а результат выводился в alert
+const age = Number(prompt(`Введи возраст`));
+console.log(age, typeof age)
+const age_2 = 18;
+const age_3 = 60;
+switch (true) {
+    case isNaN(age): {
```
Comment (anatoly-karpovich) [inline JS_lesson4/Task3.js]: Same as for previous task, i belive that isNaN check should be performed before switch

**Review by anatoly-karpovich — CHANGES_REQUESTED** [review]

File: JS_lesson4/Task2.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        }
+        case Number(age) > age_3: { 
+            console.log(`Keep calm and look Culture channel`);
+            break;
+        }
+        default: {
+            console.log("Technical work")
+            break; 
+        }
+    }
```
Comment (VladimirLQA) [inline JS_lesson4/Task2.js]: What is the purpose of declaring a string value in the `age` constant and then converting it to a number?

Pay attention to the description of the 2nd task - we only need to check the type of the value, if the value type is not a number, throw an error

Please provide a solution to the 2nd problem using the `if...else` statement.

Add a separate solution for the 3rd problem

File: JS_lesson4/Task3.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+// Преобразовать Task 2* таким образом, чтобы age принимался через prompt в привязанной вёрстке, а результат выводился в alert
+const age = Number(prompt(`Введи возраст`));
```
Comment (VladimirLQA) [inline JS_lesson4/Task3.js]: No need to convert input to a number, we should check it with provided conditions

File: JS_lesson4/Task2.js (hunk 1 above)
Comment (VladimirLQA) [inline JS_lesson4/Task2.js]: For 3-rd task add checks for `falsy` values

**Review by VladimirLQA — CHANGES_REQUESTED** [review]

File: JS_lesson4/Task3.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+// Преобразовать Task 2* таким образом, чтобы age принимался через prompt в привязанной вёрстке, а результат выводился в alert
+const age = prompt(`Введи возраст`);
+const age_2 = 18;
+const age_3 = 60;
+if (isNaN(age)) {
+    alert('Parameter is not a number!');
+    throw new Error('Parameter is not a number!'); 
+}
+switch (true) {
```
Comment (anatoly-karpovich) [inline JS_lesson4/Task3.js:9]: Seems like you need to convert age to number (cause prompt returns string) just bedore switch statement:
age = +age

Cause now you are using string value comparing with numbers

**Review by anatoly-karpovich — APPROVED** [review]

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**